scripts/globals/spells/healing_spell.lua

```diff
@@ -0,0 +1,190 @@
+
+    if castersWeather == xi.magic.singleWeatherStrong[element] then
+        weatherBonus = 0.10
+    elseif castersWeather == xi.magic.singleWeatherWeak[element] then
+        weatherBonus = -0.10
+    elseif castersWeather == xi.magic.doubleWeatherStrong[element] then
+        weatherBonus = 0.25
+    elseif castersWeather == xi.magic.doubleWeatherWeak[element] then
+        weatherBonus = -0.25
+    end
+
+    return xi.spells.healing.handleDayWeatherBonus(caster, weatherBonus, iridescence, element)
+end
+
+xi.spells.healing.getEquipBonuses = function(caster)
+    return math.min(1 + math.floor((caster:getMod(xi.mod.CURE_POTENCY) / 100) + (caster:getMod(xi.mod.CURE_POTENCY_II) / 100) + caster:getMod(xi.mod.CURE_POTENCY_BONUS)), 1.60)
+end
+
+xi.spells.healing.getAbilityBonus = function(caster, isWhiteMagic)
+    local bonus = 1
+
+    if caster:hasStatusEffect(xi.effect.RAPTURE) and isWhiteMagic then
+        bonus = bonus * (1.5 + caster:getMod(xi.mod.RAPTURE_AMOUNT) / 100)
+        caster:delStatusEffect(xi.effect.RAPTURE)
+    end
+
+    bonus = utils.ternary(caster:hasStatusEffect(xi.effect.DIVINE_SEAL), bonus * 2, bonus)
+
+    return bonus
+end
+
+xi.spells.healing.handleAfflatusSolace = function(caster, target, final)
+    if caster:isPC() and caster:hasStatusEffect(xi.effect.AFFLATUS_SOLACE) and not target:hasStatusEffect(xi.effect.STONESKIN) then
+        local bonuses = { [11186] = 0.30, [11086] = 0.35 }
+        local body = caster:getEquipID(xi.slot.BODY)
+        local stoneskin = math.floor((final * utils.ternary(bonuses[body], bonuses[body], 0.25)) * 1 + caster:getMerit(xi.merit.ANIMUS_SOLACE) / 100)
+
+        target:addStatusEffect(xi.effect.STONESKIN, stoneskin, 0, 25, 0, 0, 1)
+    end
+end
+
+xi.spells.healing.applyCasterBonuses = function(caster, final, element, isWhiteMagic)
+    local equipBonuses = xi.spells.healing.getEquipBonuses(caster)
+    local dwBonus = xi.spells.healing.getDayWeatherBonus(caster, element)
+    local abilityBonus = xi.spells.healing.getAbilityBonus(caster, isWhiteMagic)
+
+    return math.floor(((final * equipBonuses) * dwBonus) * abilityBonus)
+end
+
+xi.spells.healing.doHealingSpell = function(caster, target, spell, isWhiteMagic)
+    local final        = 0
+    local spellId      = spell:getID()
+    local power        = xi.spells.healing.calculatePower(caster)
+    local healingTable = xi.spells.healing.getHealTable(power, spellId)
+    local base         = utils.clamp(((power / 2) / healingTable.rate) + healingTable.constant, healingTable.minCap, healingTable.maxCap)
+    local healingspell = true
+
+    if xi.magic.isValidHealTarget(caster, target) then
+        final = xi.spells.healing.applyCasterBonuses(caster, base, spell:getElement(), isWhiteMagic)
+        final = (final + (final * target:getMod(xi.mod.CURE_POTENCY_RCVD))) * xi.settings.main.CURE_POWER
+        xi.spells.healing.handleAfflatusSolace(caster, target, final)
+    else
+        if target:isUndead() then
```

Comment:
isValidHealTarget returns true for mobs, I don't know if this code is reachable? I must be confused.

Reply:
isValidTarget checks for allegiance matching the player allegiance hence how it hits this code.